plugins/org.jboss.tools.openshift.cdk.server/src/org/jboss/tools/openshift/cdk/server/core/internal/adapter/CDKServer.java

```diff
@@ -43,6 +43,9 @@
 	public static final String PROP_USER_ENV_VAR = "org.jboss.tools.openshift.cdk.server.core.internal.adapter.CDKServer.env.user";
 	public static final String PROP_PASS_ENV_VAR = "org.jboss.tools.openshift.cdk.server.core.internal.adapter.CDKServer.env.pass";
 
+	public static final String PROP_SKIP_REG = "org.jboss.tools.openshift.cdk.server.core.internal.adapter.CDKServer.skip.registration";
```

Comment:
Why do we have 2 distincts flags ? I know minishift has 2 different flags for stop and start commands and for config but I think this is a design flaw.

Reply:
I was reading up on the usecase, and it seems that a user may want to start it (and register) and then during shutdown, not unregister.  Then switch both to dont register / don't unregister... and then finally, at the end,  switch to dont-register and do-unregister. 

So basically, after looking at and running through the workflows, I decided two flags were necessary because the users will most likely want to change them independently.